src/payments/sisp.test.js

```diff
@@ -0,0 +1,267 @@
+          } catch (error) {
+            expect(error[0]).toEqual(expected);
+          }
+        });
+
+        it('should return error if SISP_POS_ID is not the same as added in .env', async () => {
+          process.env.SISP_POS_ID=70051;
+          process.env.SISP_POS_AUT_CODE='123456789A';
+          process.env.SISP_WEBHOOK_URL='https://www.chuva.io/payment-response';
+
+          const referenceId = '1:3:testForm';
+          const total = 89;
+
+          const result = await sisp.generatePaymentForm(referenceId, total);
+
+          const expected = successData.replace(/\s/g, '');
+
+          expect(result.replace(/\s/g, '')).not.toEqual(expected);
+        });
+
+        it('should return error if SISP_POS_AUT_CODE is not added to .env', async () => {
+          process.env.SISP_POS_ID=90051;
+          process.env.SISP_WEBHOOK_URL='https://www.chuva.io/payment-response';
+
+          const referenceId = '1:3:testForm';
+          const total = 89;
+
+          const expected = undefined;
+
+          try {
+            await sisp.generatePaymentForm(referenceId, total);
+          } catch (error) {
+            expect(error[0]).toEqual(expected);
+          }
+        });
+
+        it('should return error if SISP_POS_AUT_CODE is not the same as added in .env', async () => {
+          process.env.SISP_POS_ID=90051;
+          process.env.SISP_POS_AUT_CODE='123456789B';
+          process.env.SISP_WEBHOOK_URL='https://www.chuva.io/payment-response';
+
+          const referenceId = '1:3:testForm';
+          const total = 89;
+
+          const result = await sisp.generatePaymentForm(referenceId, total);
+
+          const expected = successData.replace(/\s/g, '');
+
+          expect(result.replace(/\s/g, '')).not.toEqual(expected);
+        });
+
+        it('should return error if SISP_WEBHOOK_URL is not added to .env', async () => {
+          process.env.SISP_POS_ID=90051;
+          process.env.SISP_POS_AUT_CODE='123456789A';
+
+          const referenceId = '1:3:testForm';
+          const total = 89;
+
+          const expected = undefined;
+
+          try {
+            await sisp.generatePaymentForm(referenceId, total);
+          } catch (error) {
+            expect(error[0]).toEqual(expected);
+          }
+        });
+
+        it('should return error if SISP_WEBHOOK_URL is not the same as added in .env', async () => {
+          process.env.SISP_POS_ID=90051;
+          process.env.SISP_POS_AUT_CODE='123456789A';
+          process.env.SISP_WEBHOOK_URL='https://www.chuva.io/payment-respons';
+
+          const referenceId = '1:3:testForm';
+          const total = 89;
+
+          const result = await sisp.generatePaymentForm(referenceId, total);
+
+          const expected = successData.replace(/\s/g, '');
+
+          expect(result.replace(/\s/g, '')).not.toEqual(expected);
+        });
+
+        it('should return error if SISP_POS_ID and SISP_POS_AUT_CODE is not the same as added in .env', async () => {
```

Comment:
I think this is not needed because this two test above has already tested it for each of SISP_POS_ID and SISP_POS_AUT_CODE
- 'should return error if SISP_POS_AUT_CODE is not the same as added in .env'
- 'should return error if SISP_POS_ID is not the same as added in .env'

Reply:
This is to make sure it has been well tested, and these two unit tests are different, just see the description.